### DISCUSSION TITLE
Can I use slint in no X-Windows Embedded devices?

### DISCUSSION THREAD

**Discussion by jmjoy** [slint-ui/slint#3970] [Q&A] [answered]
I have a no X-Windows (no X11 or Wayland) Embedded devices, I found some app is built on SDL2, and run well.

I build a slint example, and put it in the device, and run, but it failed, with the error message "no $DISPLAY or $WAYLAND_DISPLAY ...". 

So, can I use slint in no X-Windows Embedded devices?

**Comment by ogoffart** [comment, marked as answer]
Yes you can:  You can enable the linuxkms backend.
This has to be done by enabling the backend-linuxkms feature (and you can disable other backends to save compile time and binary size).  And use the `SLINT_BACKEND=linuxkms` env variable at runtime.

More info: https://slint.dev/releases/1.3.0/docs/slint/src/advanced/backends_and_renderers and https://slint.dev/releases/1.3.0/docs/slint/src/advanced/backend_linuxkms

**Reply by jmjoy** [reply]
Thanks, `slint` is a great project.
